pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -1190,6 +1194,138 @@ func (s *sqlEntityServer) List(ctx context.Context, r *entity.EntityListRequest)
 	return rsp, err
 }
 
+func (s *sqlEntityServer) listDeleted(ctx context.Context, r *entity.EntityListRequest) (*entity.EntityListResponse, error) {
+	if err := s.Init(); err != nil {
+		return nil, err
+	}
+
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if user == nil {
+		return nil, fmt.Errorf("missing user in context")
+	}
+
+	rr := &entity.ReadEntityRequest{
+		WithBody:   r.WithBody,
+		WithStatus: r.WithStatus,
+	}
+
+	fields := s.getReadFields(rr)
+
+	entityQuery := selectQuery{
+		dialect:  s.dialect,
+		fields:   fields,
+		from:     "entity_history", // the table
+		args:     []any{},
+		limit:    r.Limit,
+		offset:   0,
+		oneExtra: true, // request one more than the limit (and show next token if it exists)
+	}
+
+	entityQuery.addWhere("action", entity.Entity_DELETED)
+
+	// TODO fix this
+	// entityQuery.addWhere("namespace", user.OrgID)
+
+	if len(r.Group) > 0 {
+		entityQuery.addWhereIn("group", r.Group)
+	}
+
+	if len(r.Resource) > 0 {
+		entityQuery.addWhereIn("resource", r.Resource)
+	}
+
+	if len(r.Key) > 0 {
+		where := []string{}
+		args := []any{}
+		for _, k := range r.Key {
+			key, err := entity.ParseKey(k)
+			if err != nil {
+				return nil, err
+			}
+
+			args = append(args, key.Group, key.Resource)
+			whereclause := "(" + s.dialect.Quote("group") + "=? AND " + s.dialect.Quote("resource") + "=?"
+			if key.Namespace != "" {
+				args = append(args, key.Namespace)
+				whereclause += " AND " + s.dialect.Quote("namespace") + "=?"
+			}
+			if key.Name != "" {
+				args = append(args, key.Name)
+				whereclause += " AND " + s.dialect.Quote("name") + "=?"
+			}
+			whereclause += ")"
+
+			where = append(where, whereclause)
+		}
+
+		entityQuery.addWhere("("+strings.Join(where, " OR ")+")", args...)
+	}
+
+	// Folder guid
+	if r.Folder != "" {
+		entityQuery.addWhere("folder", r.Folder)
+	}
+
+	// if we have a page token, use that to specify the first record
+	continueToken, err := GetContinueToken(r)
+	if err != nil {
+		return nil, err
+	}
+	if continueToken != nil {
+		entityQuery.offset = continueToken.StartOffset
+	}
+
+	if len(r.Labels) > 0 {
```

Comment:
i'm trying to parse the differences between the two list functions. it looks like this if block is one of them. are the other two major differences the table `entity_history` and `entityQuery.addWhere("action", entity.Entity_DELETED)`?